swap_meet/vendor.py

```diff
@@ -1,2 +1,65 @@
+
 class Vendor:
-    pass
\ No newline at end of file
+    def __init__(self, inventory=None):
+        if inventory is None:
+            inventory = []
+        self.inventory = inventory
+
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+    
+    def remove(self, item):
+        try: 
+            self.inventory.remove(item)
+            return item
+        except ValueError:
+            return False
+        #  ValueError will be raised if item not in self.inventory
+
+    def get_by_category(self, category):
+        return [item for item in self.inventory if item.category == category]
+
+    def swap_items(self, other, my_item, their_item):
+        if my_item not in self.inventory or their_item not in other.inventory:
+            return False    
```

Comment:
I notice that you switched from a more EAFP approach in `remove` to a LBYL approach here. This is a good call. In `remove` there's only one thing that can fail, and it either happened or not. Here, there are two removes that could fail. And if the second failed, we would need to do work to undo the first operation that didn't fail. So here, checking that the everything will work successfully ahead of winds up being a more manageable approach.